### DISCUSSION TITLE
Offline Disk Integration

### DISCUSSION THREAD

**Discussion by kassah** [spacedriveapp/spacedrive#107] [Feature Request]
**Is your feature request related to a problem? Please describe.**
A lot of photographers and other media professionals keep copies of videos and photos in offline hard disks. Enabled by hard drive docks that allow easy swapping.

**Describe the solution you'd like**
It would be really nice if spacedrive could also manage indexing, redundancy and previews of offline drives. When you go for a file that was stored on external drive in your collection, it would ask for one of the drives the file exists on to be plugged in, listing the possibilities. A major part that I desire is that it not matter which computer you're on as long as you have access to the drive itself. Bonus points for keeping track of the location of drives, and allowing for redundancy rules around separate storage location of the drives (set of drives at home, set of drives at office or friends home/safety deposit box).

**Describe alternatives you've considered**
Labeling hard drives, and trying to find the right one. Manually updated spreadsheet.

**Additional context**
This would allow for photo/video libraries that don't have to always be online or in cloud. Allowing efficient, cost effective archival storage, especially for large files.

**Comment by jamiepine** [comment]
Absolutely, this has always been one of my key motivations for Spacedrive, to search offline hard drives. This already works in the app today, not as gracefully as it will eventually. But, you can add a location for an external drive, it will scan it. once you unplug the drive the directory structure, file metadata and thumbnail images will still exist in the app.

**Reply by regaldude** [reply]
Hi,

Will the offline integration also include the ability to write data to drives that may not be available at the time. Like temporarily stashing the files elsewhere till the drive comes online again.

**Reply by jamiepine** [reply]
A file saved or moved to an offline node could cache locally until the node comes back online, then transfer the data as soon as it comes back online. Would be a simple feature to add on top of the existing architecture, only question would be how to handle the U/X of that. A "pending actions" and/or action queue UI could be a way forward on this.

**Reply by regaldude** [reply]
This would be an amazing functionality to have for offline integration.

A "Pending Uploads" screen/tab might be helpful if there are a lot of pending items. Having a "Status" attribute in the explorer [a la OneDrive](https://s14633.pcdn.co/wp-content/uploads/2021/02/onedrive_sharedfolder_fileexplorer-980x501.jpg) can also be helpful for getting the pending info at a glance. Just my 2 cents.
